pkgs/applications/editors/vim/macvim.nix

```diff
@@ -1,55 +1,73 @@
-{ stdenv, fetchFromGitHub, ncurses, gettext
-, pkgconfig, python, ruby, tcl, perl, luajit
+{ stdenv, fetchFromGitHub, runCommand, ncurses, gettext
+, pkgconfig, cscope, ruby, tcl, perl, luajit
 , darwin
+
+, usePython27 ? false
+, python27 ? null, python37 ? null
 }:
 
+let
+  python = if usePython27
+           then { pkg = python27; name = "python"; }
+           else { pkg = python37; name = "python3"; };
+in
+assert python.pkg != null;
+
+let
+  # Building requires a few system tools to be in PATH.
+  # Some of these we could patch into the relevant source files (such as xcodebuild and
+  # qlmanage) but some are used by Xcode itself and we have no choice but to put them in PATH.
+  # Symlinking them in this way is better than just putting all of /usr/bin in there.
+  buildSymlinks = runCommand "macvim-build-symlinks" {} ''
+    mkdir -p $out/bin
+    ln -s /usr/bin/xcrun /usr/bin/xcodebuild /usr/bin/tiffutil /usr/bin/qlmanage $out/bin
+  '';
+in
+
 stdenv.mkDerivation rec {
   name = "macvim-${version}";
 
-  version = "7.4.909";
+  version = "8.1.1517";
 
   src = fetchFromGitHub {
     owner = "macvim-dev";
     repo = "macvim";
-    rev = "75aa7774645adb586ab9010803773bd80e659254";
-    sha256 = "0k04jimbms6zffh8i8fjm2y51q01m5kga2n4djipd3pxij1qy89y";
+    rev = "snapshot-156";
+    sha256 = "17plmqcn49gqwr1km77mkxflrg0f4sn06r3n0fbxa8zcz9zmb1q2";
   };
 
   enableParallelBuilding = true;
 
-  nativeBuildInputs = [ pkgconfig ];
+  nativeBuildInputs = [ pkgconfig buildSymlinks ];
   buildInputs = [
-    gettext ncurses luajit ruby tcl perl python
+    gettext ncurses cscope luajit ruby tcl perl python.pkg
   ];
 
-  patches = [ ./macvim.patch ];
+  patches = [ ./macvim.patch ./macvim-sparkle.patch ];
 
+  # The sparkle patch modified the nibs, so we have to recompile them
   postPatch = ''
-    substituteInPlace src/MacVim/mvim --replace "# VIM_APP_DIR=/Applications" "VIM_APP_DIR=$out/Applications"
-
-    # Don't create custom icons.
-    substituteInPlace src/MacVim/icons/Makefile --replace '$(MAKE) -C makeicns' ""
-    substituteInPlace src/MacVim/icons/make_icons.py --replace "dont_create = False" "dont_create = True"
-
-    # Full path to xcodebuild
-    substituteInPlace src/Makefile --replace "xcodebuild" "/usr/bin/xcodebuild"
+    for nib in MainMenu Preferences; do
+      /usr/bin/ibtool --compile src/MacVim/English.lproj/$nib.nib/keyedobjects.nib src/MacVim/English.lproj/$nib.nib
```

Comment:
This could cause issues when /usr/bin/ibtool doesn't exist. Probably we should follow the Hydra builds to see. I think a ```darwin.ibtool``` might be useful to give a nice error message when this doesn't exist.

Reply:
What version of macOS does Hydra run on, and what's the minimum version of macOS that Nix supports? `/usr/bin/ibtool` should exist on any version of macOS from the past few years at least. I suppose this could be `/usr/bin/xcrun ibtool` instead if we need to support a version of macOS that predates the introduction of the full set of build tools into `/usr/bin`.